__tests__/github.test.ts

```diff
@@ -0,0 +1,92 @@
+const fakeContext = {
+  sha: "workflow-sha",
+  repo: {
+    owner: "owner",
+    repo: "repo",
+  },
+};
+describe("github", () => {
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+  });
+
+  it("creates GitHubContext from push event", () => {
+    jest.doMock("@actions/github", () => ({
+      context: {
+        ...fakeContext,
+        eventName: "push",
+      },
+    }));
+    const github = require("../src/github");
```

Comment:
This is the only way I could figure out to stub the `context` property on the `@actions/github` module without angering the build, because context` is a read-only property of a specific type.